### PR TITLE
LPS-148186 & LPS-148188 &  LPS-148189 [CP 2.0] - Setup Analytics Cloud Page for Product Activation

### PR DESCRIPTION
**Pull request by claricebrasil**
### How it was developed? 
We had to create a new objects and query both to add the Analytics Cloud Workspace and Incident Report Contacts. Also it was created a new modal to confirm when the user sent an e-mail. For that we had to create a logic to change the type of modal so it will not show one modal above the other modal.

Ps: the allowed email domains was moved to another LPS.

### CODE REVIEW

**Review by igordmf — APPROVED** [review]
Good job!

**Review by iagochaves — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js [hunk 1]
```diff
@@ -8,40 +8,82 @@
...
+				},
+			});
+
+			if (bannedDomainsItems?.length) {
+				setBannedOwnerEmailDomain(bannedDomainsItems[0].domain);
+			}
+		}
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [bannedDomainsItems, debouncedOwnerEmail]);
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js:76]: Totally agree with you, @iagochaves and I even did the hook, but I couldn't make it work. Maybe you could help me improve this? :)

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/utils/validations.form.js [hunk 2]
```diff
@@ -35,22 +35,28 @@ const isValidEmail = (value, bannedEmailDomains) => {
 	}
 };
 
+const isValidEmailDomain = (value, bannedEmailDomains) => {
+	if (bannedEmailDomains.includes(value.split('@')[1])) {
+		return 'Domain not allowed.';
+	}
+};
+
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/utils/validations.form.js:43]: Yes, this one is just for the domain and not for the whole email validation. I can think in another variable name. Thank you!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js [hunk 3]
```diff
@@ -89,76 +185,77 @@ const SetupAnalyticsCloudPage = ({
...
-									helper="Lowercase letters and numbers only. Project IDs cannot be changed."
-									label="Workspace Name"
-									name="activations.workspaceName"
-									placeholder="superbank1"
-									required
-									type="text"
-									validations={[
-										(value) => maxLength(value, 255),
-										(value) => isLowercaseAndNumbers(value),
-									]}
```
Comment (joanagrego) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js:118]: Maybe you should put this value in a variable like `MAX_LENGTH = 255.`, what do you think?

**Review by joanagrego — APPROVED** [review]
Good job!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js [hunk 4]
```diff
@@ -51,21 +93,67 @@ const SetupAnalyticsCloudPage = ({
...
+	}, [analyticsDataCenterLocations, hasDisasterRecovery]);
+
+	useEffect(() => {
+		const hasTouched = !Object.keys(touched).length;
+		const hasError = Object.keys(errors).length;
+
+		setBaseButtonDisabled(hasTouched || hasError);
+	}, [touched, errors]);
+
+	const sendEmail = async () => {
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js:120]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js [hunk 5]
```diff
@@ -51,21 +93,67 @@ const SetupAnalyticsCloudPage = ({
...
+					return client.mutate({
+						mutation: addIncidentReportAnalyticsCloud,
+						variables: {
+							IncidentReportContactAnalyticsCloud: {
+								analyticsCloudWorkspaceId,
+								emailAddress: email,
+							},
+							scopeKey: Liferay.ThemeDisplay.getScopeGroupId(),
+						},
+					});
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js:151]: done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js [hunk 6]
```diff
@@ -75,12 +163,20 @@ const SetupAnalyticsCloudPage = ({
...
 					>
 						{leftButton}
 					</Button>
 				),
-				middleButton: <Button displayType="primary">Submit</Button>,
+				middleButton: (
+					<Button
+						disabled={baseButtonDisabled}
+						displayType="primary"
+						onClick={sendEmail}
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js:175]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js (hunk 3 above)
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js:118]: Hey, Joana!! Nice, it's done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js [hunk 7]
```diff
@@ -169,37 +266,19 @@ const SetupAnalyticsCloudPage = ({
...
-						</Button>
-					</>
+					</ClayForm.Group>
 				)}
 			/>
+
+			<Button
+				className="btn-outline-primary ml-3 my-2 rounded-xs"
+				disabled={baseButtonDisabled}
+				onClick={() => setBaseButtonDisabled(true)}
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.js:276]: It should not be here, I already fix it. Thank you!!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/utils/validations.form.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (iagochaves) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/utils/validations.form.js:43]: Hmm, but the other validation is about the whole scope of the email, I think that makes sense to use that too.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js [hunk 8]
```diff
@@ -194,18 +173,60 @@ export const getDXPCloudEnvironment = gql`
 	}
 `;
 
+export const addAnalyticsCloudWorkspace = gql`
+	mutation addAnalyticsCloudWorkspace(
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js:177]: Hey Iago, I followed the DXP and Team Members patterns for the name of this query. What do you think? Maybe we can change all query names?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js [hunk 9]
```diff
@@ -226,6 +247,23 @@ export const addAdminDXPCloud = gql`
 	}
 `;
 
+export const addIncidentReportAnalyticsCloud = gql`
+	mutation addIncidentReportAnalyticsCloud(
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js:251]: Same thing here

**Pull request closed without merging**